zebra-chain/src/transparent/serialize.rs

```diff
@@ -78,20 +78,33 @@ fn parse_coinbase_height(
         // Blocks 17 through 128 exclusive encode block height with the `0x01` opcode.
         // The Bitcoin encoding requires that the most significant byte is below 0x80.
         (Some(0x01), len) if len >= 2 && data[1] < 0x80 => {
-            Ok((Height(data[1] as u32), CoinbaseData(data.split_off(2))))
+            let h = data[1] as u32;
+            if h < 128 {
```

Comment:
Likely Bug: 1..16 are covered by the shorter encoding above, so they are not allowed in this longer encoding.

```suggestion
            if (17..128).contains(&h) {
```

Reply:
Thank you for this one it seems i missed it. Fixed it in https://github.com/ZcashFoundation/zebra/pull/3129/commits/8c2064526a8858554ec3f91ae8297012f76fa78b